### PR TITLE
IUM-415: New rule to migrate the new user profile root attributes.

### PR DESCRIPTION
**Pull request by gustavojucoskifernandes**
A new rule to migrate the new user profile root attributes.

### CODE REVIEW

File: src/rules/migrate_root_attributes.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+    for (const key in fieldMapping) {
+      generateUserPayloadField(userMetadata, payload, key, fieldMapping[key]);
+    }
+
+    return payload;
+  }
+
+  function generateUserPayloadField(userMetadata, payload, rootField, metadataField) {
+    if (userMetadata[metadataField]) {
```
Comment (siacomuzzi) [inline src/rules/migrate_root_attributes.js]:
```suggestion
    if (typeof userMetadata[metadataField] === 'string') {
```

File: src/rules/migrate_root_attributes.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+      domain: auth0.domain,
+      token: auth0.accessToken
+    });
+
+    management.updateUser(
+      { id: user.user_id }, generateUserPayload(user), function (err, user) {
+        if ( err ) {
+          cb(err);
+        } else {
+          cb(null, user, context);
```
Comment (siacomuzzi) [inline src/rules/migrate_root_attributes.js]: did you check if it's ok to return a new `user` object (api2's result) instead of the original one?

Comment (gustavojucoskifernandes) [reply src/rules/migrate_root_attributes.js]: Yes, I did, it returns the updated user.

Comment (gustavojucoskifernandes) [reply src/rules/migrate_root_attributes.js]: I changed the code to use the original one.

**Comment by siacomuzzi** [comment]
LGTM

**Comment by lzychowski** [comment]
hey, @gustavojucoskifernandes can we please include some integration tests for this rule?

**Comment by gustavojucoskifernandes** [comment]
> hey, @gustavojucoskifernandes can we please include some integration tests for this rule?

Done

**Pull request merged**